packages/components/nodes/vectorstores/Couchbase/Couchbase.ts

```diff
@@ -169,8 +178,28 @@ class Couchbase_VectorStores implements INode {
                 if (!textKey || textKey === '') couchbaseConfig.textKey = 'text'
                 if (!embeddingKey || embeddingKey === '') couchbaseConfig.embeddingKey = 'embedding'
 
-                await CouchbaseVectorStore.fromDocuments(finalDocs, embeddings, couchbaseConfig)
-                return { numAdded: finalDocs.length, addedDocs: finalDocs }
+                if (recordManager) {
+                    const vectorStore = await CouchbaseVectorStore.initialize(embeddings, couchbaseConfig)
+
+                    await recordManager.createSchema()
+
+                    const res = await index({
+                        docsSource: finalDocs,
+                        recordManager,
+                        vectorStore,
+                        options: {
+                            cleanup: recordManager?.cleanup,
+                            sourceIdKey: recordManager?.sourceIdKey ?? 'source',
+                            vectorStoreName: [bucketName, scopeName, collectionName].join('_')
```

Comment:
@prajwal-pai77 @HenryHengZJ I'm wondering how to handle the vectorStoreName key for indexing, any advice ?

should we include `indexName` ?